Firestore/core/test/unit/bundle/bundle_reader_test.cc

```diff
@@ -352,18 +371,33 @@ TEST_F(BundleReaderTest, ReadsQueryAndDocumentWithUnexpectedOrder) {
       VerifyFullBundleParsed(reader, "bundle-1", testutil::Version(6000004000));
 
   EXPECT_EQ(elements.size(), 5);
-  VerifyDocumentMetadataEquals(
-      *reinterpret_cast<BundledDocumentMetadata*>(elements[0].get()),
-      DocumentMetadata1());
-  VerifyDocumentEncodesToOriginal(
-      *reinterpret_cast<BundleDocument*>(elements[1].get()), Document1());
-  VerifyNamedQueryEncodesToOriginal(
-      *reinterpret_cast<NamedQuery*>(elements[2].get()), LimitQuery());
-  VerifyDocumentMetadataEquals(
-      *reinterpret_cast<BundledDocumentMetadata*>(elements[3].get()),
-      DocumentMetadata2());
-  VerifyDocumentEncodesToOriginal(
-      *reinterpret_cast<BundleDocument*>(elements[4].get()), Document2());
+  {
+    SCOPED_TRACE("DocumentMetadata1");
+    VerifyDocumentMetadataEquals(
+        *static_cast<BundledDocumentMetadata*>(elements[0].get()),
+        DocumentMetadata1());
+  }
+  {
+    SCOPED_TRACE("Document1");
+    VerifyDocumentEncodesToOriginal(
+        *static_cast<BundleDocument*>(elements[1].get()), Document1());
+  }
+  {
+    SCOPED_TRACE("LimitQuery");
```

Comment:
Note: these `SCOPED_TRACE`s should stay (except `LimitQuery`, which isn't strictly necessary -- but it's ok to keep for symmetry).

Reply:
Ack.